worlds/soe/__init__.py

```diff
@@ -208,14 +208,68 @@ def stage_assert_generate(cls, world):
+        for loc in _locations:
+            spheres.setdefault(min(2, len(loc.requires)), {}).setdefault(loc.type, []).append(
+                SoELocation(self.player, loc.name, self.location_name_to_id[loc.name], r,
+                            loc.difficulty > max_difficulty))
+
+        # location balancing data
+        trash_fills: typing.Dict[int, typing.Dict[int, typing.Tuple[int, int, int, int]]] = {
+            0: {pyevermizer.CHECK_GOURD: (20, 40, 40, 40)},  # remove up to 40 gourds from sphere 1
+            1: {pyevermizer.CHECK_GOURD: (70, 90, 90, 90)},  # remove up to 90 gourds from sphere 2
+        }
+
+        # mark some as excluded based on numbers above
+        for trash_sphere, fills in trash_fills.items():
+            for typ, counts in fills.items():
+                count = counts[self.world.difficulty[self.player].value]
+                for location in self.world.random.sample(spheres[trash_sphere][typ], count):
+                    location.progress_type = LocationProgressType.EXCLUDED
+                    # TODO: do we need to set an item rule?
+
+        def sphere1_blocked_items_rule(item):
+            if isinstance(item, SoEItem):
+                # disable certain items in sphere 1
+                if item.name in {"Gauge", "Wheel"}:
+                    return False
+                # and some more for non-easy, non-mystery
+                if self.world.difficulty[item.player] not in (Difficulty.option_easy, Difficulty.option_mystery):
+                    if item.name in {"Laser Lance", "Atom Smasher", "Diamond Eye"}:
+                        return False
+            return True
+
+        for locations in spheres[0].values():
+            for location in locations:
+                add_item_rule(location, sphere1_blocked_items_rule)
+
+        # make some logically late(r) bosses priority locations to increase complexity
+        if self.world.difficulty[self.player] == Difficulty.option_mystery:
+            late_count = self.world.random.randint(0, 2)
+        else:
+            late_count = self.world.difficulty[self.player].value
+        late_bosses = ("Tiny", "Aquagoth", "Megataur", "Rimsala",
+                       "Mungola", "Lightning Storm", "Magmar", "Volcano Viper")
+        late_locations = self.world.random.sample(late_bosses, late_count)
+
+        # add locations to the world
         r = Region('Ingame', RegionType.Generic, 'Ingame', self.player, self.world)
-        r.locations = [SoELocation(self.player, loc.name, self.location_name_to_id[loc.name], r)
-                       for loc in _locations]
+        for sphere in spheres.values():
+            for locations in sphere.values():
+                for location in locations:
+                    r.locations.append(location)
+                    if location.name in late_locations:
+                        location.progress_type = LocationProgressType.PRIORITY
```

Comment:
similar idea as exclude before; might be better to add the name to world.priority_locations[player].value

Reply:
there i *think* i don't want it in the spoiler, because it's "just" balancing (which is why i did it this way), but if you think it *should* be in the spoiler, let me know.